hazelcast/src/test/java/com/hazelcast/jet/core/metrics/JobLifecycleMetricsTest.java

```diff
@@ -114,17 +123,20 @@ public void jobSuspendedThenResumed() {
         //init
         Job job = hzInstances[0].getJet().newJob(streamingPipeline());
         assertJobStatusEventually(job, RUNNING);
+        assertTrueEventually(() -> assertJobStatusMetric(job, RUNNING));
 
         //when
         job.suspend();
         assertJobStatusEventually(job, SUSPENDED);
+        assertTrueEventually(() -> assertJobStatusMetric(job, SUSPENDED));
 
         //then
         assertTrueEventually(() -> assertJobStats(1, 1, 1, 0, 0));
 
         //when
         job.resume();
         assertJobStatusEventually(job, RUNNING);
+        assertTrueEventually(() -> assertJobStatusMetric(job, RUNNING));
```

Comment:
we need also a test with `isStoreMetricsAfterJobCompletion` option

Reply:
@burakgok comment is still unresolved.

Reply:
we test that additionally in `jobCancelled` and `executionRelatedMetrics` tests. Is it enough? We do not have a test for job failed on it's own (not cancelled by the user).

Reply:
c7f813e9099c3c90f4ee81abd4a403f983458900